<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TerminalWidget</class>
 <widget class="QWidget" name="TerminalWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>677</width>
    <height>371</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Simple Cross-Platform Serial Terminal</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/utilities-terminal.png</normaloff>:/utilities-terminal.png</iconset>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,1,1">
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Connection</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QGridLayout" name="gridLayout" columnstretch="0,1,0">
        <item row="1" column="1">
         <widget class="QComboBox" name="comboBoxBaudrate">
          <property name="editable">
           <bool>true</bool>
          </property>
          <property name="maxVisibleItems">
           <number>15</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Baudrate</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Port</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="comboBoxPort">
          <property name="editable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Data bits</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QComboBox" name="comboBoxDatabits"/>
        </item>
        <item row="3" column="1">
         <widget class="QComboBox" name="comboBoxParity"/>
        </item>
        <item row="4" column="1">
         <widget class="QComboBox" name="comboBoxStopbits"/>
        </item>
        <item row="5" column="1">
         <widget class="QComboBox" name="comboBoxFlowcontrol"/>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Parity</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Stop bits</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Flow control</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="pbUpdatePorts">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="resources.qrc">
            <normaloff>:/view-refresh.png</normaloff>:/view-refresh.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="lbPortInfo">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>10</width>
          <height>10</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QPushButton" name="pushButtonConnect">
          <property name="text">
           <string>Connect</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonDisconnect">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Disconnect</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Tx</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,0,0,1,0,0">
      <item>
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Local echo</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="plainTextEditTxPrevious">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="acceptDrops">
         <bool>false</bool>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pbClearTx">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>clear</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Input</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="plainTextEditTx"/>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLabel" name="lbTxInterpret">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Interpret as</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbTxText">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Text will be sent as typed in the input field</string>
          </property>
          <property name="text">
           <string>Text</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbTxHex">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Text is interpreted as hexadecimal numbers in in the range [0,FF]. Use space for separating.</string>
          </property>
          <property name="text">
           <string>Hex</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbTxBin">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Text is interpreted as binary numbers in in the range [0,11111111]. Use space for separating.</string>
          </property>
          <property name="text">
           <string>Bin</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,0,0">
        <item row="0" column="0">
         <widget class="QPushButton" name="pushButtonSend">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Send</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QCheckBox" name="cbSendLF">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Append line feed character (0x0A) to input text</string>
          </property>
          <property name="text">
           <string>LF</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QCheckBox" name="cbSendCR">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Append carriage return character (0x0D) to input text</string>
          </property>
          <property name="text">
           <string>CR</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QCheckBox" name="cbSendOnReturn">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Send on Return</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Rx</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QPlainTextEdit" name="plainTextEditRx">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="acceptDrops">
         <bool>false</bool>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pbClearRx">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>clear</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="lbRxDisplay">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Display as</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbRxText">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Received data will be displayed as ASCII text</string>
          </property>
          <property name="text">
           <string>Text</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbRxHex">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Received data will be displayed as hexadecimal numbers</string>
          </property>
          <property name="text">
           <string>Hex</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbRxBin">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Received data will be displayed as binary numbers</string>
          </property>
          <property name="text">
           <string>Bin</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
